playbooks/robusta_playbooks/persistent_volume_actions.py

```diff
@@ -0,0 +1,170 @@
+import json
+from robusta.api import action, PersistentVolumeEvent, PersistentVolumeClaim, FileBlock, RobustaPod, ObjectMeta, PodSpec, Volume, PersistentVolumeClaimVolumeSource, Container, VolumeMount, MarkdownBlock, Finding, FindingSource, FindingType
+from kubernetes import client
+
+
+@action
+def volume_analysis(event: PersistentVolumeEvent):
+    """
+    This action shows you the files present on your persistent volume
+    """
+    function_name = "volume_analysis"
+    # https://docs.robusta.dev/master/developer-guide/actions/findings-api.html
+    finding = Finding(
+        title=f"Persistent Volume content",
+        source=FindingSource.MANUAL,
+        aggregation_key=function_name,
+        finding_type=FindingType.REPORT,
+        failure=False,
+    )
+
+    # Get persistent volume data the object contains data related to PV like metadata etc
+    pv = event.get_persistentvolume()
+    pv_claimref = pv.spec.claimRef
+
+    try:
+
+        if pv_claimref != None:
+            # Do this if there is a PVC attached to PV
+            pvc_obj = PersistentVolumeClaim.readNamespacedPersistentVolumeClaim(
+                name=pv_claimref.name, namespace=pv_claimref.namespace).obj
+            pod_obj = get_pod_related_to_pvc(pvc_obj, pv)
+            if (pod_obj != None):
+                # Do this if a Pod is using PVC
+
+                pod = RobustaPod.find_pod(
+                    pod_obj.metadata.name, pod_obj.metadata.namespace)
+                volume_mount_name = None
+
+                # Find name of the mounted volume on pod
+                for volume in pod.spec.volumes:
+                    if volume.persistentVolumeClaim.claimName == pv_claimref.name:
+                        volume_mount_name = volume.name
+                        break
+
+                # Use name of the mounted volume to find the correct volume mount
+                container_found_flag = False
+                container_volume_mount = None
+                for container in pod.spec.containers:
+                    if (container_found_flag):
+                        break
+                    if(container.volumeMounts):
```

Comment:
I think you can remove that if, because volumeMount is a list
So, if it's empty nothing happens in the for loop

Reply:
Resolved. 
Check if the current change is ok. Feel free to tell me anything.